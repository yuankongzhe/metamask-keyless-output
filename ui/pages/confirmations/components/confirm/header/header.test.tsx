import React from 'react';

import { fireEvent, waitFor } from '@testing-library/react';
import mockState from '../../../../../../test/data/mock-state.json';
import { renderWithProvider } from '../../../../../../test/jest';
import configureStore from '../../../../../store/store';

import Header from './header';

const render = () => {
  const store = configureStore({
    metamask: {
      ...mockState.metamask,
    },
    confirm: {
      currentConfirmation: {
        msgParams: {
          from: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
        },
      },
    },
  });

  return renderWithProvider(<Header />, store);
};

describe('Header', () => {
  it('should match snapshot', async () => {
    const { container } = render();
    expect(container).toMatchSnapshot();
  });
  it('contains network name and account name', () => {
    const { getByText } = render();
    expect(getByText('Test Account')).toBeInTheDocument();
    expect(getByText('Chain 5')).toBeInTheDocument();
  });
  it('contains account info icon', async () => {
    const { getByLabelText } = render();
    expect(getByLabelText('Account details')).toBeInTheDocument();
  });
  it('shows modal when account info icon is clicked', async () => {
    const { getByLabelText, queryByTestId } = render();
    expect(queryByTestId('account-details-modal')).not.toBeInTheDocument();
    const accountInfoIcon = getByLabelText('Account details');
    fireEvent.click(accountInfoIcon);
    await waitFor(() => {
      expect(queryByTestId('account-details-modal')).toBeInTheDocument();
    });
  });
});
