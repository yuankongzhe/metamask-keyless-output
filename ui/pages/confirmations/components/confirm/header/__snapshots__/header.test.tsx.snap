// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <div
    class="mm-box confirm_header__wrapper mm-box--display-flex mm-box--justify-content-space-between mm-box--align-items-center"
  >
    <div
      class="mm-box mm-box--padding-4 mm-box--display-flex mm-box--align-items-flex-start"
    >
      <div
        class="mm-box mm-box--margin-top-2 mm-box--display-flex"
      >
        <div
          class=""
        >
          <div
            class="identicon"
            style="height: 32px; width: 32px; border-radius: 16px;"
          >
            <div
              style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 32px; height: 32px; display: inline-block; background: rgb(250, 58, 0);"
            >
              <svg
                height="32"
                width="32"
                x="0"
                y="0"
              >
                <rect
                  fill="#18CDF2"
                  height="32"
                  transform="translate(-1.04839350379394 -3.3042840694604987) rotate(328.9 16 16)"
                  width="32"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#035E56"
                  height="32"
                  transform="translate(-18.298461708832043 10.5924618717486) rotate(176.2 16 16)"
                  width="32"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#F26602"
                  height="32"
                  transform="translate(16.667842018223922 -14.205139722997082) rotate(468.9 16 16)"
                  width="32"
                  x="0"
                  y="0"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network confirm_header__avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
        >
          C
        </div>
      </div>
      <div
        class="mm-box mm-box--margin-inline-start-4"
      >
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-default"
        >
          Test Account
        </p>
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative"
        >
          Chain 5
        </p>
      </div>
    </div>
    <div
      class="mm-box mm-box--padding-4 mm-box--display-flex mm-box--align-items-flex-end"
    >
      <div
        class="mm-box mm-box--display-flex mm-box--justify-content-flex-end"
        style="align-self: flex-end;"
      >
        <div>
          <div
            aria-describedby="tippy-tooltip-1"
            class=""
            data-original-title="Account details"
            data-tooltipped=""
            style="display: inline;"
            tabindex="0"
          >
            <button
              aria-label="Account details"
              class="mm-box mm-button-icon mm-button-icon--size-md mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-icon-default mm-box--background-color-transparent mm-box--rounded-lg"
            >
              <span
                class="mm-box mm-icon mm-icon--size-md mm-box--display-inline-block mm-box--color-inherit"
                style="mask-image: url('./images/icons/info.svg');"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
